extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/nonblocking/NonBlockingScheduledMethodTest.java

```diff
@@ -0,0 +1,131 @@
+import io.quarkus.test.QuarkusUnitTest;
+import io.smallrye.common.annotation.NonBlocking;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.Context;
+
+public class NonBlockingScheduledMethodTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest test = new QuarkusUnitTest()
+            .withApplicationRoot(root -> root.addClasses(Jobs.class, JobWasExecuted.class));
+
+    @Test
+    public void testVoid() throws InterruptedException {
+        assertTrue(Jobs.VOID_LATCH.await(5, TimeUnit.SECONDS));
+        assertTrue(Jobs.VOID_ON_EVENT_LOOP.get());
+        assertTrue(Jobs.SUCCESS_LATCH.await(5, TimeUnit.SECONDS));
+        assertEvents("every_void");
+    }
+
+    @Test
+    public void testUni() throws InterruptedException {
+        assertTrue(Jobs.UNI_LATCH.await(5, TimeUnit.SECONDS));
+        assertTrue(Jobs.UNI_ON_EVENT_LOOP.get());
+        assertTrue(Jobs.SUCCESS_LATCH.await(5, TimeUnit.SECONDS));
+        assertEvents("every_uni");
+    }
+
+    @Test
+    public void testCompletionStage() throws InterruptedException {
+        assertTrue(Jobs.CS_LATCH.await(5, TimeUnit.SECONDS));
+        assertTrue(Jobs.CS_ON_EVENT_LOOP.get());
+        assertTrue(Jobs.SUCCESS_LATCH.await(5, TimeUnit.SECONDS));
+        assertEvents("every_cs");
+    }
+
+    private void assertEvents(String id) {
+        for (SuccessfulExecution exec : Jobs.events) {
+            if (exec.getExecution().getTrigger().getId().equals(id)) {
+                return;
+            }
+        }
+        fail("No SuccessfulExecution event fired for " + id + ": " + Jobs.events);
+    }
+
+    static class Jobs {
+
+        // jobs executed
+        static final CountDownLatch VOID_LATCH = new CountDownLatch(1);
+        static final CountDownLatch UNI_LATCH = new CountDownLatch(1);
+        static final CountDownLatch CS_LATCH = new CountDownLatch(1);
+
+        // jobs executed on the event loop
+        static final AtomicBoolean VOID_ON_EVENT_LOOP = new AtomicBoolean();
+        static final AtomicBoolean UNI_ON_EVENT_LOOP = new AtomicBoolean();
+        static final AtomicBoolean CS_ON_EVENT_LOOP = new AtomicBoolean();
+
+        // sucessfull events
+        static final CountDownLatch SUCCESS_LATCH = new CountDownLatch(3);
+        static final List<SuccessfulExecution> events = new CopyOnWriteArrayList<>();
+
+        static void onSuccess(@Observes SuccessfulExecution event) {
+            events.add(event);
+            SUCCESS_LATCH.countDown();
+        }
+
+        @NonBlocking
+        @Scheduled(every = "0.5s", identity = "every_void", skipExecutionIf = JobWasExecuted.class)
+        void everySecond() {
+            VOID_ON_EVENT_LOOP.set(Context.isOnEventLoopThread());
+            VOID_LATCH.countDown();
+        }
+
+        @Scheduled(every = "0.5s", identity = "every_uni", skipExecutionIf = JobWasExecuted.class)
+        Uni<Void> everySecondUni() {
+            UNI_ON_EVENT_LOOP.set(Context.isOnEventLoopThread());
```

Comment:
This should be scheduled on an event loop and running on a duplicated context. 

I think that all scheduled execution should use a separate duplicated context, to avoid leaking data between 2 scheduled executions. 

So, it would mean that, when executing a non-blocking scheduled operation, you need something like:

```java
Vertx vertx = ...;
// ...
Context context = VertxContext.createNewDuplicatedContext(vertx);
CompletableFuture<Void> future = new CompletableFuture<>()

// For CS:
context.runOnContext(() -> action.execute().handle(i, f) > {
   if (f != null) { future.completeExceptionally(f));
   else { future.complete(null));
});

// For uni

context.runOnContext(() -> action.execute().subscribeWith(
item -> future.complete(null), failure -> future.completeExceptionally(future)
);

return future;
```

Reply:
The missing part is the duplicated context thingy.

Reply:
I don't see a static method `VertxContext.createNewDuplicatedContext()` that would accept a `Vertx` param?

Reply:
@cescoffier ^ ?

Reply:
Hm, is it the `io.smallrye.common.vertx.VertxContext.getOrCreateDuplicatedContext(Vertx)` method that should be used here...?

Reply:
I've modified the assertions to use `VertxContext.isOnDuplicatedContext()` as well.